src/common/templates/_hpa.tpl

```diff
@@ -1,13 +1,13 @@
 {{/*
 Create Horizontal Pod Autoscaler Configuration
 Usage example:
-{{- include "harnesscommon.hpa.renderHPA" . (dict "ctx" .  "kind" "deployment") }}
+{{- include "harnesscommon.hpa.renderHPA" . (dict "ctx" .  "kind" "deployment" "targetRefNameOverride" "custom-target-name") }}
 */}}
 {{- define "harnesscommon.hpa.renderHPA" -}}
 {{- $ := .ctx }}
+{{- $targetRefName := default (default $.Chart.Name $.Values.nameOverride | trunc 63 | trimSuffix "-") .targetRefNameOverride }}
```

Comment:
Since this is defined as `targetRefNameOverride`, let's only override that
```
{{- $serviceName := default $.Chart.Name $.Values.nameOverride | trunc 63 | trimSuffix "-" }}
{{- $targetRefName := default $serviceName .targetRefNameOverride }}
```